sql/core/src/main/scala/org/apache/spark/sql/columnar/ColumnAccessor.scala

```diff
@@ -118,17 +112,15 @@ private[sql] object ColumnAccessor {
     dup.getInt()
 
     dataType match {
-      case IntegerType => new IntColumnAccessor(dup)
-      case LongType => new LongColumnAccessor(dup)
-      case FloatType => new FloatColumnAccessor(dup)
-      case DoubleType => new DoubleColumnAccessor(dup)
       case BooleanType => new BooleanColumnAccessor(dup)
       case ByteType => new ByteColumnAccessor(dup)
       case ShortType => new ShortColumnAccessor(dup)
+      case IntegerType | DateType => new IntColumnAccessor(dup)
```

Comment:
It looks to me that we don't need date and timestamp column accessor/builder, did I miss something here? @liancheng